src/parser/jobs/brd/modules/Snapshots.tsx

```diff
@@ -0,0 +1,197 @@
+import React from 'react'
+import {Trans} from '@lingui/react'
+import {t} from '@lingui/macro'
+import {Table} from 'semantic-ui-react'
+import Module, {dependency} from 'parser/core/Module'
+import {getDataBy} from 'data'
+import {CastEvent} from 'fflogs'
+import STATUSES from 'data/STATUSES'
+import {SNAPSHOT_BLACKLIST} from 'parser/jobs/brd/modules/SnapshotBlacklist'
+import ACTIONS from 'data/ACTIONS'
+import {ActionLink, StatusLink} from 'components/ui/DbLink'
+import {NormalisedDamageEvent} from 'parser/core/modules/NormalisedEvents'
+import Combatants from 'parser/core/modules/Combatants'
+import {AbilityEvent} from 'fflogs'
+import Util from './Util'
+
+const SNAPSHOTTERS = [
+	ACTIONS.IRON_JAWS.id,
+	ACTIONS.CAUSTIC_BITE.id,
+	ACTIONS.STORMBITE.id,
+	ACTIONS.VENOMOUS_BITE.id,
+	ACTIONS.WINDBITE.id,
+]
+
+const DOT_STATUSES = [
+	STATUSES.CAUSTIC_BITE.id,
+	STATUSES.STORMBITE.id,
+	STATUSES.VENOMOUS_BITE.id,
+	STATUSES.WINDBITE.id,
+]
+
+const PERSONAL_STATUSES = [
+	STATUSES.MEDICATED.id,
+	STATUSES.RAGING_STRIKES.id,
+]
+
+interface Snapshot {
+	// The event that applied the DoT(s).
+	snapEvent: CastEvent
+
+	// All statuses observed at the time of casting the snapshot.
+	statusIDs: number[]
+
+	// DoT ticks that occurred under the current snapshot.
+	ticks: NormalisedDamageEvent[]
+}
+
+export default class Snapshots extends Module {
+	static handle = 'snapshots'
+	static title = t('brd.snapshots.title')`Snapshots`
+	static debug = true
+
+	@dependency private combatants!: Combatants
+	@dependency private util!: Util
+
+	private snapshots: Snapshot[] = []
+	private currentSnapshot?: Snapshot
+
+	private targets = new Map<string, AbilityEvent[]>()
+
+	protected init() {
+		this.addEventHook('cast', {by: 'player', abilityId: SNAPSHOTTERS}, this._onSnapshot)
+		this.addEventHook('normaliseddamage', {by: 'player', abilityId: DOT_STATUSES}, this._onDotTick)
+		this.addEventHook(['applydebuff', 'refreshdebuff'], this._onApply)
+		this.addEventHook('removedebuff', this._onRemove)
+	}
+
+	private _onApply(event: AbilityEvent) {
+		const targetKey = `${event.targetID}-${event.targetInstance}`
+
+		if (!this.targets.has(targetKey)) {
+			this.targets.set(targetKey, [])
+		}
+
+		const target = this.targets.get(targetKey)
+		if (target) {
+			const index = target.findIndex(statusEvent =>
+				statusEvent.ability.guid === event.ability.guid)
+
+			if (index === -1) {
+				target.push(event)
+			}
+		}
+	}
+
+	private _onRemove(event: AbilityEvent) {
+		const targetKey = `${event.targetID}-${event.targetInstance}`
+		const targetStatuses = this.targets.get(targetKey)
+
+		if (targetStatuses) {
+			const index = targetStatuses.findIndex(status =>
+				status.ability.guid === event.ability.guid)
+
+			if (index > -1) {
+				targetStatuses.splice(index, 1)
+			}
+		}
+	}
+
+	private _getStatuses(event: CastEvent) {
```

Comment:
Prob more of a nit since TS is good at stuff but with a more complex return type like this, may be good to explicitly declare it.